### PR TITLE
Add system dark theme

### LINKED ISSUE #341: automatic dark/light mode switch on android 11 and lower
### Describe your suggested feature

ik theres monet support but theres no system auto dark and light support for older android versions

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by epireyn**
This adds an option to set the theme accordingly to the system's. Closes #341.

### CODE REVIEW

File: app/src/main/res/values-pl/array.xml [hunk 1]
```diff
@@ -252,13 +252,15 @@
 
 
     <string-array name="themes_names">
+        <item>System</item>
         <item>Ciemny</item>
         <item>Szary</item>
         <item>Amoled</item>
         <item>Flashbang</item>
-        <item>Material You</item>
+        <item>Material You (Android 12+)</item>
```
Comment (fire-light42) [inline app/src/main/res/values-pl/array.xml:260]: Why "(Android 12+)" that is simply redundant.

**Review by fire-light42 — CHANGES_REQUESTED** [review]

**Comment by epireyn** [comment]
Indeed, I noticed afterwards but forgot to revert it.

File: app/src/main/java/com/lagradost/cloudstream3/ui/settings/SettingsUI.kt [hunk 2]
```diff
@@ -88,10 +88,9 @@ class SettingsUI : PreferenceFragmentCompat() {
         getPref(R.string.app_theme_key)?.setOnPreferenceClickListener {
             val prefNames = resources.getStringArray(R.array.themes_names).toMutableList()
             val prefValues = resources.getStringArray(R.array.themes_names_values).toMutableList()
-
-            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) { // remove monet on android 11 and less
+            val removeFunction = { text: String ->
```
Comment (C10udburst) [inline app/src/main/java/com/lagradost/cloudstream3/ui/settings/SettingsUI.kt:91]: maybe removeTheme would be more descriptive?

**Review by fire-light42 — CHANGES_REQUESTED** [review]
This makes system the default because of `settingsManager.getString(getString(R.string.primary_color_key), prefValues.first())`

"Dark" should be the default, so I need you to change it.

**Comment by Luna712** [comment]
> This makes system the default because of `settingsManager.getString(getString(R.string.primary_color_key), prefValues.first())`
> 
> "Dark" should be the default, so I need you to change it.

@fire-light42

If the intent is to always default to dark, wouldn't it make sense to instead force that to Dark as a default rather than `.first()` it seems that would be less prone to potential changes messing it up. Anyway it is up to you, that is just my personal opinion but there may also be a reason why not to do that. But doing it also just seems like it would better indicate the intended default as well.

**Comment by epireyn** [comment]
> > This makes system the default because of `settingsManager.getString(getString(R.string.primary_color_key), prefValues.first())`
> > "Dark" should be the default, so I need you to change it.
> 
> @fire-light42
> 
> If the intent is to always default to dark, wouldn't it make sense to instead force that to Dark as a default rather than `.first()` it seems that would be less prone to potential changes messing it up. Anyway it is up to you, that is just my personal opinion but there may also be a reason why not to do that. But doing it also just seems like it would better indicate the intended default as well.

It would be the best, however it requires hard-coding the value. If it was to change, the themes wouldn't work at all. I reorder the list, it's no big issue.

File: app/src/main/res/values-tr/array.xml [hunk 3]
```diff
@@ -277,13 +277,15 @@
...
     <string-array name="themes_names">
+        <item>Sistem</item>
         <item>Koyu</item>
         <item>Gri</item>
         <item>Amoled</item>
         <item>Flaş Bombası</item>
         <item>Material You</item>
     </string-array>
     <string-array name="themes_names_values">
+        <item>System</item>
```
Comment (fire-light42) [inline app/src/main/res/values-tr/array.xml:288]: You forgot to reorder all arrays

**Review by fire-light42 — CHANGES_REQUESTED** [review]

**Review by fire-light42 — APPROVED** [review]

**Comment by fire-light42** [comment]
Now more people can be flashbanged by the white theme 🤩

**Pull request merged**